CMakeLists.txt

```diff
@@ -265,6 +265,9 @@ if(SFML_OS_MACOSX)
     set(XCODE_TEMPLATES_ARCH "\$(NATIVE_ARCH_ACTUAL)")
 endif()
 
+# set the output directory for SFML DLLs and executables
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
```

Comment:
Should this be done no a per-target basis with `set_target_properties`?

Reply:
I can't think of any reason we'd want to do that.

Reply:
Any reasons to set it globally instead of target specific then?

Reply:
Must less code is required plus it keeps the door open for easily converting this into an option in the future which I think may be necessary to make sure the CMake template can easily put all these DLLs alongside the executables that users write.

Reply:
Not sure that really convinced me (beyond less code not really being an argument). I'm no expert, but I've been told on multiple occasions not to use global modifiers, as they can also easily pollute consuming projects and instead configure things on the target themselves.

Maybe @friendlyanon has some insights on this?

Reply:
`CMAKE_RUNTIME_OUTPUT_DIRECTORY` is a **user-controllable** cache variable. It shouldn't be hardcoded in project code.

Ideally, SFML would just use a package manager where this is taken care for you or you modify `PATH` manually. Windows lacks an RPATH mechanism, so what can you do.

Reply:
> ...not to use global modifiers

This is generally sound advice. The problem comes from people who use commands like `include_directories` when they need to specify `INTERFACE` requirements since those global modifiers only specify private details of a target. This leads to people writing targets that fail to specify all of their usage requirements. That's not a problem in our case since we don't want this to be a usage requirement of our targets.

> as they can also easily pollute consuming projects and instead configure things on the target themselves.

I'm fairly certain that as-written, setting this variable will not change where downstream projects place their own binaries so there is no pollution to be had. I do need to make this variable a CACHE variable so it can more easily be changed by users.

Reply:
Yeah currently this will affect all targets so we should just set it on the SFML targets that need it

Reply:
I want this to affect all targets.

Reply:
even non-SFML ones? this means if I use fetchcontent for SFML my exe is going to go into SFML's build directory (assuming it's a cached value as you say)

Reply:
Ah you're right. When setting this as a cache variable (which is something I think we need to do) then that means projects using SFML via `add_subdirectory` will have their own executables placed into build/_deps/sfml-build/bin which is definitely not desirable.

I'll experiment with setting them on a per-target basis to see if that lets us achieve both the convenient behavior we want while also letting downstream users pick the directory where all the SFML DLLs go. I think there's room to improve on the status quo but I'm still unsure if the ideal solution I'd like is possible.

Reply:
I'd imagine a better solution would be to make sure openAL (and any other required) dlls are part of the RUNTIME_DLL list. 

Not something I've done before but expect it should be doable

Reply:
Have since proved we can get openAL on the runtime DLL list so can probably just bail on this